ex_webrtc/lib/ex_webrtc/peer_connection_handler/metrics.ex

```diff
@@ -0,0 +1,264 @@
+      ),
+      Telemetry.Metrics.last_value(
+        "remote_candidate.address",
+        event_name: @remote_candidate_event,
+        measurement: :address
+      ),
+      Telemetry.Metrics.last_value(
+        "remote_candidate.protocol",
+        event_name: @remote_candidate_event,
+        measurement: :protocol
+      )
+    ]
+  end
+
+  @spec emit_inbound_packet_event(
+          ExRTP.Packet.t(),
+          MediaStreamTrack.id(),
+          MediaStreamTrack.rid() | nil,
+          TelemetryMetrics.label()
+        ) :: :ok
+  def emit_inbound_packet_event(packet, webrtc_track_id, rid, telemetry_label) do
+    track_label =
+      if is_nil(rid) do
+        [track_id: webrtc_track_id]
+      else
+        [track_id: "#{webrtc_track_id}:#{rid}"]
+      end
+
+    telemetry_label = telemetry_label ++ track_label
+
+    TelemetryMetrics.execute(
+      @packet_received_event,
+      %{bytes: byte_size(packet.payload)},
+      %{},
+      telemetry_label
+    )
+
+    :ok
+  end
+
+  @spec emit_outbound_packet_event(
+          ExRTP.Packet.t(),
+          MediaStreamTrack.id(),
+          TelemetryMetrics.label()
+        ) :: :ok
+  def emit_outbound_packet_event(packet, webrtc_track_id, telemetry_label) do
+    telemetry_label = telemetry_label ++ [track_id: webrtc_track_id]
+
+    TelemetryMetrics.execute(
+      @packet_sent_event,
+      %{bytes: byte_size(packet.payload)},
+      %{},
+      telemetry_label
+    )
+
+    :ok
+  end
+
+  @spec register_events(TelemetryMetrics.label()) :: :ok
+  def register_events(telemetry_label) do
+    Enum.each(
+      [
+        @inbound_rtp_event,
+        @packet_received_event,
+        @outbound_rtp_event,
+        @packet_sent_event
+      ],
+      &TelemetryMetrics.register(&1, telemetry_label)
+    )
+
+    :ok
+  end
+
+  @spec emit_from_rtc_stats(rtc_stats_report(), TelemetryMetrics.label()) :: :ok
+  def emit_from_rtc_stats(metrics, telemetry_label) do
+    Enum.each(metrics, fn {_id, entry} -> handle_entry(entry, telemetry_label) end)
+  end
+
+  defp handle_entry(%{type: :inbound_rtp} = entry, telemetry_label) do
+    telemetry_label = telemetry_label ++ [track_id: entry.track_identifier]
+
+    TelemetryMetrics.execute(
+      @inbound_rtp_event,
+      Map.take(entry, [
+        :packets_received,
+        :bytes_received,
+        :nack_count,
+        :pli_count,
+        :markers_received,
+        :codec
+      ]),
+      %{},
+      telemetry_label
+    )
+  end
+
+  defp handle_entry(%{type: :outbound_rtp} = entry, telemetry_label) do
+    telemetry_label = telemetry_label ++ [track_id: entry.track_identifier]
+
+    TelemetryMetrics.execute(
+      @outbound_rtp_event,
+      Map.take(entry, [
+        :packets_sent,
+        :bytes_sent,
+        :nack_count,
+        :pli_count,
+        :markers_sent,
+        :codec,
+        :retransmitted_packets_sent,
+        :retransmitted_bytes_sent
+      ]),
+      %{},
+      telemetry_label
+    )
+  end
```

Comment:
How does it work in the case of simulcast?

Reply:
We track each variant separately (the `track_identifier` has a variant in its name)